src/core/manager.c

```diff
@@ -4717,16 +4713,9 @@ char* manager_taint_string(const Manager *m) {
 
         assert(m);
 
-        const char* stage[13] = {};
+        const char* stage[11] = {};
         size_t n = 0;
 
-        if (m->taint_usr)
-                stage[n++] = "split-usr";
-
-        _cleanup_free_ char *usrbin = NULL;
-        if (readlink_malloc("/bin", &usrbin) < 0 || !PATH_IN_SET(usrbin, "usr/bin", "/usr/bin"))
-                stage[n++] = "unmerged-usr";
```

Comment:
i think this taint should really state. it detects the nonsense gentoo is doing, with symlinking all individual binaries between /usr/bin/ and /bin. And that's really stupid and should continue to be reported as a taint.

Gentoo really just opted for an approach that only makes the usrmerge a bad thing without providing anything good...

Reply:
Are you sure they are doing that? I thought only SUSE tried that (and walked back)

Reply:
Have you actually tried it?

Reply:
you know, i don't need to stick a fork in my hand to understand that it's a terrible approach to scratch an itch.

Reply:
No, that's *not* what is happening. I think you may have misread that section - it's describing how the _script_ does the transformation. The final result is a symlinked directory and *not* symlinks per-files.

Reply:
right that's how I understood it was implemented too, thanks for confirming

Reply:
Right, the reason I asked that question was as a hint that you seem to have drastically misunderstood the entire article.

It's describing a live migration process. Every file is carefully compat-symlinked one by one after being moved to prevent live system instability. Once the entire directory is fully migrated, those symlinks are purged and replaced by the directory symlinks everyone is familiar with.

I see no point in avoiding per-file symlinks during the halfway stage??? Not everyone has the luxury of being able to migrate to usrmerge while offline and booted to a recovery USB.

Reply:
Hey, just chiming in as the author of Gentoo's merge-usr migration script and the original author of that wiki page.

The symlinks for individual binaries are just an interim state while the script is performing the migration. They are just used to ensure that commands don't go missing while we are moving files around.

At the very end of the script, the /bin and /lib directories are replaced with symlinks, and the old directories containing the interim binary symlinks are removed.

Also, @poettering, feel free to ping me in the future if you see something that doesn't look right. Always open to feedback.